elixir/apps/domain/lib/domain/accounts/account/query.ex

```diff
@@ -1,46 +1,103 @@
 defmodule Domain.Accounts.Account.Query do
   use Domain, :query
-  alias Domain.Validator
 
   def all do
-    from(account in Domain.Accounts.Account, as: :account)
+    from(accounts in Domain.Accounts.Account, as: :accounts)
   end
 
   def not_deleted(queryable \\ all()) do
-    where(queryable, [account: account], is_nil(account.deleted_at))
+    where(queryable, [accounts: accounts], is_nil(accounts.deleted_at))
   end
 
   def not_disabled(queryable \\ not_deleted()) do
-    where(queryable, [account: account], is_nil(account.disabled_at))
+    where(queryable, [accounts: accounts], is_nil(accounts.disabled_at))
   end
 
-  def by_id(queryable \\ not_deleted(), id)
+  def by_id(queryable, id)
 
   def by_id(queryable, {:in, ids}) do
-    where(queryable, [account: account], account.id in ^ids)
+    where(queryable, [accounts: accounts], accounts.id in ^ids)
   end
 
   def by_id(queryable, id) do
-    where(queryable, [account: account], account.id == ^id)
+    where(queryable, [accounts: accounts], accounts.id == ^id)
   end
 
   def by_stripe_customer_id(queryable, customer_id) do
     where(
       queryable,
-      [account: account],
-      fragment("?->'stripe'->>'customer_id' = ?", account.metadata, ^customer_id)
+      [accounts: accounts],
+      fragment("?->'stripe'->>'customer_id' = ?", accounts.metadata, ^customer_id)
     )
   end
 
-  def by_slug(queryable \\ not_deleted(), slug) do
```

Comment:
I've removed `\\ queryable()` everywhere so that our domain modules are more explicit, before you had to go to query and see if it will return all, or deleted, or active, or _whatever_ records. Now you have to explicitly set that as a first stage when a query is built.